roles/translator/src/upstream_sv2/upstream.rs

```diff
@@ -674,7 +705,7 @@ impl ParseUpstreamMiningMessages<Downstream, NullDownstreamMiningSelector, NoRou
 
         self.target
             .safe_lock(|t| *t = m.maximum_target.to_vec())
-            .unwrap();
+            .map_err(|_e| RolesLogicError::PoisonLock)?;
```

Comment:
Why not pass the `e` into the PoisonLock?

Reply:
The compiler is very finicky with the safe_lock error type since it contains information about the `MutexGuard` and a sync mutex guard cant be sent across a threads. For some reason it wont even let me convert the error to a string and pass in the string so the best thing I could think of is to make a PoisonLock variant that we can use as a replacement